src/_imagingcms.c

```diff
@@ -201,8 +201,8 @@ cms_transform_new(cmsHTRANSFORM transform, char *mode_in, char *mode_out) {
 
     self->transform = transform;
 
-    strcpy(self->mode_in, mode_in);
-    strcpy(self->mode_out, mode_out);
+    strncpy(self->mode_in, mode_in, 8);
+    strncpy(self->mode_out, mode_out, 8);
```

Comment:
`strncpy` doesn’t null-terminate oversized entries.

```suggestion
    strncpy(self->mode_in, mode_in, 8);
    self->mode_in[7] = '\0';
    strncpy(self->mode_out, mode_out, 8);
    self->mode_out[7] = '\0';
```

But are these even used? The Python `ImageCmsTransform` wrapper has `inputMode` and `outputMode` attributes already, and they work normally. Is there a need to preserve a slightly different `.transform.inputMode`?

Reply:
Probably not, see #7931 for a suggestion to remove them.